File: ios/Nexo/Models/TeamMember.swift
//
//  TeamMember.swift
//  Nexo
//
//  Created by Atzmi, Dan on 25/04/2026.
//

import Foundation

struct TeamMember: Identifiable {
    let id: String  // Firebase UID
    var firstName: String
    var lastName: String
    var email: String
    var role: UserRole

    var fullName: String {
        "\(firstName) \(lastName)".trimmingCharacters(in: .whitespaces)
    }
}

File: android/app/src/main/java/com/nexo/app/domain/model/TeamMember.kt
package com.nexo.app.domain.model

/**
 * Mirrors `gyms/{gymId}/team/{uid}` — see FIRESTORE_SCHEMA.md. `id` is the
 * Firebase Auth UID. Only ever holds `role == owner` or `role == coach`
 * entries — members live in `gyms/{gymId}/members` instead.
 *
 * NOTE: [profilePicBase64] isn't a field on the real `team` document (only
 * `role`/`firstName`/`lastName`/`email`/`addedAt` are, per
 * FIRESTORE_SCHEMA.md and iOS's `fetchTeam`) — it's always `null` when
 * parsed from Firestore. Kept as an optional field anyway so [AvatarView]
 * has one consistent signature across `Member`/`GymMember`/`TeamMember`.
 */
data class TeamMember(
    val id: String,
    val fullName: String,
    val email: String,
    val role: UserRole,
    val profilePicBase64: String? = null
)
